src/Controls/tests/BindingSourceGen.UnitTests/IntegrationTests.cs

```diff
@@ -111,6 +111,338 @@ private static bool ShouldUseSetter(BindingMode mode, BindableProperty bindableP
+                        {
+                            Mode = mode,
+                            Converter = converter,
+                            ConverterParameter = converterParameter,
+                            StringFormat = stringFormat,
+                            Source = source,
+                            FallbackValue = fallbackValue,
+                            TargetNullValue = targetNullValue
+                        };
+                        bindableObject.SetBinding(bindableProperty, binding);
+                    }
+
+                    private static bool ShouldUseSetter(BindingMode mode, BindableProperty bindableProperty)
+                        => mode == BindingMode.OneWayToSource
+                            || mode == BindingMode.TwoWay
+                            || (mode == BindingMode.Default
+                                && (bindableProperty.DefaultBindingMode == BindingMode.OneWayToSource
+                                    || bindableProperty.DefaultBindingMode == BindingMode.TwoWay));
+                }
+            }
+            """,
+            result.GeneratedCode);
+    }
+
+    [Theory]
+    [InlineData(
+        """
+        // Reference Type
+        #nullable disable
+        using Microsoft.Maui.Controls;
+        using MyNamespace;
+
+        var label = new Label();
+        label.SetBinding(Label.RotationProperty, static (A a) => a.B.C);
+
+        namespace MyNamespace
+            {
+                public class A
+                {
+                    public B B { get; set; } = null!;
+                }
+
+                public class B
+                {
+                    public C C { get; set; }
+                }
+
+                public class C
+                {
+
+                }
+            }
+        """
+    )]
+    [InlineData(
+        """
+        // Nullable value type
+        #nullable disable
+        using Microsoft.Maui.Controls;
+        using MyNamespace;
+
+        var label = new Label();
+        label.SetBinding(Label.RotationProperty, static (A a) => a.B.C);
+
+        namespace MyNamespace
+            {
+                public class A
+                {
+                    public B B { get; set; } = null!;
+                }
+
+                public class B
+                {
+                    public C? C { get; set; }
+                }
+
+                public struct C
+                {
+
+                }
+            }
+        """
+    )]
+    [InlineData(
+        """
+        // Condtional access operator
+        #nullable disable
+        using Microsoft.Maui.Controls;
+        using MyNamespace;
+
+        var label = new Label();
+        label.SetBinding(Label.RotationProperty, static (A a) => a?.B.C);
+
+        namespace MyNamespace
+            {
+                public class A
+                {
+                    public B B { get; set; } = null!;
+                }
+
+                public class B
```

Comment:
Can you please add one more test case where we have `struct B`? We must avoid checking for nullability (in the case of `public B B { get; set; }`).

Reply:
Im not sure I got it. You refer to a scenario when we have non-nullable value type on the path, and we want to access one of its properties?

Reply:
I had something like this in mind:

```c#
// Condtional access operator
#nullable disable
using Microsoft.Maui.Controls;
using MyNamespace;
var label = new Label();
label.SetBinding(Label.RotationProperty, static (A a) => a?.B.C);
namespace MyNamespace
{
    public class A
    {
        public B B { get; set; }
    }
    public struct B
    {
        public C C { get; set; } = null!;
    }
    public class C
    {
    }
}
```
the output would differ in this:
```c#
setter = static (source, value) =>
{
    if (!value.HasValue)
    {
        return;
    }
    if (source is {} p0)
    {
        p0.B.C = value.Value;
    }
};
```